### PR TITLE
[MTDSA-4872] retrieve transactions request parser & validator

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: app/v1/controllers/requestParsers/RetrieveTransactionsRequestDataParser.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package v1.controllers.requestParsers
+
+import javax.inject.Inject
+import uk.gov.hmrc.domain.Nino
+import v1.controllers.requestParsers.validators.RetrieveTransactionsValidator
+import v1.models.request.retrieveTransactions.{RetrieveTransactionsParsedRequest, RetrieveTransactionsRawRequest}
+
+class RetrieveTransactionsRequestDataParser  @Inject()(val validator: RetrieveTransactionsValidator)
```
Comment (BetaDraconis) [inline app/v1/controllers/requestParsers/RetrieveTransactionsRequestDataParser.scala]: Could we rename the Parsers to all either be "RequestParser" or "RequestDataParser"? At the moment we're using a mixture of both.

File: test/v1/controllers/requestParsers/validators/FromAndToValidatorSpec.scala [hunk 2]
```diff
@@ -18,64 +18,65 @@ package v1.controllers.requestParsers.validators
...
 
-  val validator = new ListChargesValidator()
+  class Validator extends FromAndToValidator[RetrieveTransactionsRawRequest]
+  val validator = new Validator
 
   "validation" should {
     "return no errors" when {
       "passed valid raw data" in {
-        val input = ListChargesRawRequest("AA123456A", Some("2019-02-01"), Some("2019-02-02"))
+        val input = RetrieveTransactionsRawRequest("AA999999A", Some("2019-02-01"), Some("2019-02-02"))
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply test/v1/controllers/requestParsers/validators/FromAndToValidatorSpec.scala]: good idea, one sec

File: app/v1/controllers/requestParsers/validators/FromAndToValidator.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import v1.models.request.FromAndToRawData
+import v1.controllers.requestParsers.validators.validations.{DateFormatValidation, DateRangeValidation, MissingParameterValidation, NinoValidation}
+import v1.models.errors._
+
+trait FromAndToValidator[T <: FromAndToRawData] extends Validator[T] {
```
Comment (BetaDraconis) [inline app/v1/controllers/requestParsers/validators/FromAndToValidator.scala]: I like the implementation, i'm just not entirely sure about the name. I would suggest 'ListValidator' but the transactions endpoint isn't _technically_ a list endpoint. Just think it needs to be clear that we're validating the nino as well.

Comment (james-work-account) [reply app/v1/controllers/requestParsers/validators/FromAndToValidator.scala]: How about something like `ValidatorWithDateRange`?

**Pull request merged**